routers/user/changePassword.ts

```diff
@@ -0,0 +1,37 @@
+import { NextFunction, Request, Response } from 'express';
+import { ValidationError } from '../../utils/errorsHandler';
+import { myDataSource } from '../../config/database.configuration';
+import { UserPayloadData } from '../../utils/createTokens';
+import { ChangePasswordRequest } from '../../types';
+import { User } from '../../src/entities/User/User.entity';
+import bcrypt from 'bcryptjs';
+import { validatePassword } from '../../utils/validatePassword';
+
+type RequestAndPayloadUser = Request & UserPayloadData;
+
+export const changePassword = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
type for req should be RequestAndPayloadUser so in that case you don't need write in line 14 req.user as RequestAndPayloadUser;

Reply:
Jeżeli tak zrobię jak piszesz to w pliku gdzie jest wywoływana funkcja TS wywala błąd z typami.